synapseclient/models/submission.py

```diff
@@ -0,0 +1,939 @@
+    """
+
+    etag: Optional[str] = None
+    """The current eTag of the Entity being submitted. If not provided, it will be automatically retrieved."""
+
+    def fill_from_dict(
+        self, synapse_submission: dict[str, Union[bool, str, int, list]]
+    ) -> "Submission":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            synapse_submission: The response from the REST API.
+
+        Returns:
+            The Submission object.
+        """
+        self.id = synapse_submission.get("id", None)
+        self.user_id = synapse_submission.get("userId", None)
+        self.submitter_alias = synapse_submission.get("submitterAlias", None)
+        self.entity_id = synapse_submission.get("entityId", None)
+        self.version_number = synapse_submission.get("versionNumber", None)
+        self.evaluation_id = synapse_submission.get("evaluationId", None)
+        self.evaluation_round_id = synapse_submission.get("evaluationRoundId", None)
+        self.name = synapse_submission.get("name", None)
+        self.created_on = synapse_submission.get("createdOn", None)
+        self.team_id = synapse_submission.get("teamId", None)
+        self.contributors = synapse_submission.get("contributors", [])
+        self.submission_status = synapse_submission.get("submissionStatus", None)
+        self.entity_bundle_json = synapse_submission.get("entityBundleJSON", None)
+        self.docker_repository_name = synapse_submission.get(
+            "dockerRepositoryName", None
+        )
+        self.docker_digest = synapse_submission.get("dockerDigest", None)
+
+        return self
+
+    async def _fetch_latest_entity(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> dict:
+        """
+        Fetch the latest entity information from Synapse.
+
+        <https://rest-docs.synapse.org/rest/GET/entity/id.html>
+
+        If the object is a DockerRepository, this will also fetch the DockerCommit object with the latest createdOn value
+        and attach it to the final dictionary:
+
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/docker/DockerCommit.html>
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            Dictionary containing entity information from the REST API.
+
+        Raises:
+            ValueError: If entity_id is not set or if unable to fetch entity information.
+        """
+        if not self.entity_id:
+            raise ValueError("entity_id must be set to fetch entity information")
+
+        from synapseclient import Synapse
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        try:
+            entity_info = await client.rest_get_async(f"/entity/{self.entity_id}")
```

Comment:
This could be replaced by this function call https://github.com/Sage-Bionetworks/synapsePythonClient/blob/436125780767c4cecb4fc0810d3d1fe6aa72cd75/synapseclient/api/entity_services.py#L151 which will make this API call.